test/test-common.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+# vim: ts=4 sw=4 ts
+
+test_id=$(echo $0 | sed -e 's/^[^a-z]*//' | sed -e 's/.sh$//')
```

Comment:
Great feedback.  For years, I've been using sub-processes to do this work, this is much cleaner.  Addressed in commit: 4b3e33ac9eb7c8d072bc9e763424c400539352ef